actors/miner/tests/precommit_expiry.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019-2022 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use fil_actor_miner::BitFieldQueue;
+use fil_actors_runtime::runtime::Policy;
+use fvm_ipld_blockstore::Blockstore;
+use fvm_shared::clock::ChainEpoch;
+
+use std::collections::BTreeMap;
+
+mod util;
+use util::*;
+
+mod state_harness;
+use state_harness::*;
+
+const PERIOD_OFFSET: ChainEpoch = 0;
+
+#[derive(Default, Clone)]
+pub struct BitfieldQueueExpectation {
+    pub expected: BTreeMap<ChainEpoch, Vec<u64>>,
+}
+
+impl BitfieldQueueExpectation {
```

Comment:
Take a look at @jdjaustin's PR #340 which defines this as well.  They look very similar but feel free to leave comments there addressing any discrepencies.  I think you're approach might be slightly more readable.  Once you've rebased this on that work I'll take another pass at review